marklogic-data-hub/src/main/resources/ml-modules/root/data-hub/5/flow/flowRunner.sjs

```diff
@@ -176,6 +177,34 @@ function invokeBeforeMain(stepExecutionContext, contentArray) {
   }
 }
 
+/**
+ * Applies source collections to written documents if targetCollectionsAdditivity is set to true
+ * @param stepExecutionContext 
+ * @param contentArray 
+ */
+function applyTargetCollectionsAdditivity(stepExecutionContext, contentArray) {
+  
+  if (!stepExecutionContext.combinedOptions.targetCollectionsAdditivity ||
+    String(stepExecutionContext.combinedOptions.targetCollectionsAdditivity) != "true")
+  {
+    // do nothing
+    return;
+  }
+
+  // append context.originalCollections to context.collections
```

Comment:
No need for this comment - it's only correct some of the time (if collections doesn't exist, then it's a "set", not an "append") and it's easy enough to understand the code that no comment is needed.

Reply:
You can also simplify the code by avoiding the if/else:

```
if (content.context.originalCollections) {
   const collections = content.context.collections || [];
   content.context.collections = collections.concat(content.context.originalCollection);
}
```